test/integration/070_incremental_schema_tests/test_incremental_schema.py

```diff
@@ -63,6 +63,12 @@ def run_incremental_sync_remove_only(self):
         compare_target = 'incremental_sync_remove_only_target'
         self.run_twice_and_assert(select, compare_source, compare_target)
 
+    def run_incremental_sync_shorter_cols(self):
+        select = 'model_a incremental_sync_shorter_columns incremental_sync_shorter_target'
```

Comment:
@epapineau  thanks for contributing and adding the tests! I tried to run this test locally without your change in [column_helpers.sql](https://github.com/dbt-labs/dbt-core/pull/5395/files#diff-fe6d3769c671db9592b1f967fe22259cc439717adb90254e12ed5960abf46938) and the tests still passed. Do you think you can update the test case a bit so that it replicate the issue happened in #5395?

Reply:
BTW, this function won't actually run unless you have a test function that actually runs it. something like 
```python 
    @use_profile('postgres')
    def test__postgres__run_incremental_sync_shorter_cols(self):
        self.run_incremental_sync_shorter_cols()

Reply:
Thanks for the feedback @ChenyuLInx. I don't think this bug exists on postgres - I was only able to recreate on Snowflake & Redshift (didn't try on BigQuery). Am I able to pass a different test db as a value for `@use_profile()`?

Reply:
Gotcha, we don't really have another option in dbt-core, so I think we should probably just add the code here, and add a test in `dbt-snowflake` or `dbt-redshift` separately.